README.md

```diff
@@ -6,288 +6,324 @@ Code to generate and deploy the dashboard for the Cumulus API.
 
-#### <a name=dockerdiagram></a> Docker Container Service Diagram.
-![Docker Service Diagram](./ancillary/DashboardDockerServices.png)
-
-#### NGAP Sandbox Metrics Development
 
-##### Kibana and Elasticsearch access
+## Testing
 
-In order to develop features that interact with Kibana or Elasticsearch in the NGAP sandbox, you need to set up tunnels through the metric's teams bastion-host.  First you must get access to the metric's host. This will require a [NASD ticket](https://bugs.earthdata.nasa.gov/servicedesk/customer/portal/7/create/79) and permission from the metrics team.  Once you have access to the metrics-bastion-host you can get the IP addresses for the Bastion, Kibana and Elasticsearch from the metrics team and configure your `.ssh/config` file to create you local tunnels.  This configuration will open traffic to the Kibana endpoint on localhost:5601 and Elasticsearch on localhost:9201 tunneling traffic through the Bastion and Kibana machines.
+### Unit Tests
 
+```bash
+  $ npm run test
 ```
-Host metrics-bastion-host
-  Hostname "Bastion.Host.Ip.Address"
-  User ec2-user
-  IdentitiesOnly yes
-  IdentityFile ~/.ssh/your_private_bastion_key
-Host metrics-elk-tunnels
-  Hostname "Kibana.Host.IP.Address"
-  IdentitiesOnly yes
-  ProxyCommand ssh metrics-bastion-host -W %h:%p
-  User ec2-user
-  IdentityFile ~/.ssh/your_private_bastion_key
-  # kibana
-  LocalForward 5601 "Kibana.Host.IP.Address":5601
-  # elastic search
-  LocalForward 9201 "Elasticsearch.Host.IP.Address":9201
+
+### Integration & Validation Tests
+
+For the integration tests to work, you have to first run the localstack application, launch the localAPI and serve the dashboard first. Run the following commands in separate terminal sessions:
+
+Run background localstack application.
+```bash
+  $ npm run start-localstack
 ```
 
-Now you can configure you sandbox environment with these variables.
+Serve the cumulus API (separate terminal)
+```bash
+  $ npm run serve-api
+```
 
-```sh
-export ESROOT=http://localhost:9201
-export KIBANAROOT=http://localhost:5601
+Serve the dashboard web application (another terminal)
+```bash
+  $ [HIDE_PDR=false SHOW_DISTRIBUTION_API_METRICS=true ESROOT=http://example.com APIROOT=http://localhost:5001] npm run serve
 ```
 
-If the Elasticsearch machine is protected by basic authorization, the following two variables should also be set.
+If you're just testing dashboard code, you can generally run all of the above commands as a single docker-compose stack.
+```bash
+  $ npm run start-dashboard
+```
+This brings up LocalStack, Elasticsearch, the Cumulus localAPI, and the dashboard.
 
-```sh
-export ES_USER=<username>
-export ES_PASSWORD=<password>
+Run the test suite (yet another terminal window)
+```bash
+  $ npm run cypress
 ```
 
+When the cypress editor opens, click on `run all specs`.
 
 
-### Running locally in Docker
+### local API server
```

Comment:
Local or local

Reply:
Also what was the thought behind putting this stuff after the testing part? if the testing mentions using the local api

Reply:
Should be Local.  I was just thinking the Local API server shouldn't get confused with any of the normal build or the docker build stuff.  Is it referenced above?  I could link to it up there I suppose.  Do you think it should go back?